script/rsync.sh

```diff
@@ -79,6 +79,12 @@ set +e
     [ ! -L "$subfolder/.run_last" ] || rm "$subfolder/.run_last"
     ln -s -T "$(pwd)/$logdir" $subfolder/.run_last
 
+    # run additional hook script from current subfolder
+    [[ -x "$subfolder/hook_script" ]] && "$subfolder/hook_script" 2>$logdir/hook_script.err > $logdir/hook_script.out
```

Comment:
Can the hook_script have any effect on the command line that is generated and executed afterwards? How about to allow to export variables in the hook script which are read if defined by the later scripts to be able to customize the generated isos sync command(s)?

Reply:
I think the established pattern here is that scripts generate certain output and those files and then be consumed later on. So I would just have the hook script generated e.g. a file with scenario definitions and then maybe also directly the `openqa.cmd` file with the `openqa-cli` commands that will be invoked later by `rsync.sh`.

Not sure how it would work with environment variables. We cannot just source the hook script because it might also be e.g. a Python script.

Reply:
@Martchus, think that a JSON data structure would be the lingua franca for hooks to exchange information. Following up on that, the solution may go in a different direction.

@Martchus, I'm happy to chat and brainstorm in the office or during a call

Reply:
> think that a JSON data structure would be the lingua franca for hooks to exchange information. Following up on that, the solution may go in a different direction.

Note that the consumer of the hook script output is also just a script (`print_openqa.sh` if present; otherwise `rsync.sh` will just invoke `openqa.cmd` if present).  So probably it makes most sense to simply generate the required `openqa-cli` commands and write them to `openqa.cmd` and skip having `print_openqa.sh`.

It is really *not* like there's some subsequent logic in openQA that would be able to consume a JSON file. There's just the generation of `openqa-cli` commands within this trigger repo itself and so far this is just a generated shell script.

> I'm happy to chat and brainstorm in the office or during a call

We'll probably have a chance to talk in the office as of the 24th.